src/Company/Yurtici/Service.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Teknomavi\Kargo\Company\Yurtici;
+
+
+use Teknomavi\Kargo\Company\Yurtici\Helper\YKSoapClient;
+use Teknomavi\Kargo\Company\ServiceAbstract;
+use Teknomavi\Kargo\Company\ServiceInterface;
+use Teknomavi\Kargo\Response\CreateShipment;
+use Teknomavi\Kargo\Model\Package;
+use Teknomavi\Kargo\Response\PackageInfo;
+use Teknomavi\Kargo\Response\ShipmentStatus;
+
+/**
+ * Class Service
+ * @package Teknomavi\Kargo\Company\Yurtici
+ */
+class Service extends ServiceAbstract implements ServiceInterface
+{
+    private $YKClient;
+
+    function __construct($options)
+    {
+        $this->YKClient = new YKSoapClient($options);
+    }
+
+    public function sendPackages()
+    {
+        $responses = array();
+        foreach ($this->packages as $package) {
+            $res = $this->YKClient->createShipment(
+                $package->getReferenceNo(),
+                $package->getInvoiceNo(),
+                $package->getConsigneeName(),
+                $package->getConsigneeAddress(),
+                $package->getConsigneeCity(),
+                $package->getConsigneeTown(),
+                $package->getConsigneeMobilPhone(),
+                "",
+                "",
+                $package->getConsigneeEmail(),
+                "",
+                "",
+                "",
+                $package->getNumberOfPackages()
+            );
+
+            $response = new CreateShipment();
+            if ($res->ShippingOrderResultVO->outFlag == 0) {
+                $response = $response->setSuccess(true);
+                $response->setReferenceNumber($res->ShippingOrderResultVO->shippingOrderDetailVO->cargoKey);
+            } else {
+                $response = $response->setSuccess(false);
+                $response = $response->setErrorDescription($res->ShippingOrderResultVO->shippingOrderDetailVO->errMessage);
+                $response = $response->setErrorCode($res->ShippingOrderResultVO->shippingOrderDetailVO->errCode);
+            }
+            $responses[$package->getReferenceNo()] = $response;
+        }
+        return $responses;
+    }
+
+    public function addPackage(Package $package)
+    {
+        array_push($this->packages, $package);
```

Comment:
Shipment objesini addPackage fonksiyonu içinde oluşturup packages listesinde hazır objeyi tutuyoruz. Büyük listelerle sorgu yaptığımızda memory'de gereksiz obje tutmamak için böyle bir yol izliyoruz.